src/physics/linear_winds.f90

```diff
@@ -739,8 +750,9 @@ subroutine initialize_spatial_winds(domain,options,reverse)
 
                         ! call update_irregular_grid(u,v, nsq_values(j), z, domain, minimum_layer_size)
                         call linear_perturbation(u, v, exp(nsq_values(j)),                                                                      &
-                                                 domain%global_z_interface(:,z,:) - domain%global_terrain,                                      &
-                                                 domain%global_z_interface(:,z,:) - domain%global_terrain + domain%global_dz_interface(:,z,:),  &
+                                                 domain%global_z_interface(:,z,:) - domain%global_terrain,                                      &   ! z_bottom
+                                                 domain%global_z_interface(:,z,:) - domain%global_terrain + domain%global_dz_interface(:,z,:),  &   ! z_top
+                                                !  domain%global_z_interface(:,z+1,:) - domain%global_terrain ,  &   ! z_top : Maybe this is a better definition?? should be the same...
```

Comment:
the problem may be that z_interface(z+1) might not exist?

Reply:
yes, one would have to adjust the looping index, but i've removed the comment.